front/src/modules/trainschedule/components/TimetableV2/utils.ts

```diff
@@ -32,5 +32,8 @@ export const extractTagCode = (tag?: string | null) => {
 export const timetableHasInvalidTrain = (trains: TrainScheduleWithDetails[]) =>
   trains.some((train) => train.invalidReason);
 
-export const transformDateWithoutYear = (dateTime: string): string =>
-  `${dateTime.split('/')[0]}/${dateTime.split('/')[1]} ${dateTime.split('/')[2].slice(5)} `;
+export const transformDateWithoutYear = (dateTime: string): string => {
+  const parts = dateTime.split('/');
+  if (!parts[0] || !parts[1] || !parts[2]) return '';
+  return `${parts[0]}/${parts[1]} ${parts[2].slice(5)}`;
+};
```

Comment:
## Structure

If I understand correctly, this function takes a date and formats it, ignoring the year.

Weirdly, the input is a string and not a `Date` type. We should use a standard library to do this kind of operation.
Date formatting is complicated and depends on the user's region:

Examples:
- Canada: `YYYY-MM-DD`
- France: `DD/MM/YYYY`
- Italy: `DD.MM.YY`
- US: `MM-DD-YYYY`

## The fix

- Returning an empty string when the input format is invalid is unpredictable and must be documented
- We should have unit tests to avoid regression

> [!NOTE]
> Ideally, changing the behavior of a function should necessarily lead to a modification of docs and tests.

Reply:
Fixed, can you check behaviour and new implementation :)